File: backend/src/main/java/com/team05/petmeeting/domain/animal/repository/AnimalRepositoryCustom.java
package com.team05.petmeeting.domain.animal.repository;

import com.team05.petmeeting.domain.animal.entity.Animal;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AnimalRepositoryCustom {

    Page<Animal> findAnimalsWithFilter(
            String region,
            String kind,
            Integer stateGroup,
            Pageable pageable
    );


}

File: backend/src/main/java/com/team05/petmeeting/domain/animal/repository/AnimalRepositoryCustom.kt
package com.team05.petmeeting.domain.animal.repository

import com.team05.petmeeting.domain.animal.entity.Animal
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface AnimalRepositoryCustom {
    fun findAnimalsWithFilter(
        region: String?,
        kind: String?,
        stateGroup: Int?,
        pageable: Pageable
    ): Page<Animal>
}
